### PR TITLE
Feature/boilerplate generation

### PR DESCRIPTION
**Pull request by ChrysmOre**
# What
Initial boilerplate generation and refactor, add Dockerfile and reflex file

# How
Check changes make sense

# Who 
Anyone

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# Mac
+.DS_Store
+
+#VSCode
+.vscode
+
+# Intellij
+.idea/
+*.iml
+*.iws
```
Comment (redhug1) [inline .gitignore]: Add CR

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+test:
+	go test -race -cover ./...
+
+.PHONY: convey
+convey:
+	goconvey ./...
+
+.PHONY: test-component
+test-component:
+	go test -cover -coverpkg=github.com/ONSdigital/dp-cantabular-dimension-api/... -component
```
Comment (redhug1) [inline Makefile]: Add CR

File: ci/audit.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  source:
+    repository: onsdigital/dp-concourse-tools-nancy
+    tag: latest
+
+inputs:
+  - name: dp-cantabular-dimension-api
+    path: dp-cantabular-dimension-api
+
+run:
+  path: dp-cantabular-dimension-api/ci/scripts/audit.sh
```
Comment (redhug1) [inline ci/audit.yml]: Add CR

File: ci/lint.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  type: docker-image
+  source:
+    repository: golang
+    tag: latest
+
+inputs:
+  - name: dp-cantabular-dimension-api
+
+run:
+  path: dp-cantabular-dimension-api/ci/scripts/lint.sh
```
Comment (redhug1) [inline ci/lint.yml]: Add CR

File: ci/scripts/audit.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -eux
+
+pushd dp-cantabular-dimension-api
+  make audit
+popd
```
Comment (redhug1) [inline ci/scripts/audit.sh]: Add CR

File: ci/scripts/lint.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -eux
+
+pushd dp-cantabular-dimension-api
+  make lint
+popd
```
Comment (redhug1) [inline ci/scripts/lint.sh]: Add CR

File: dp-cantabular-dimension-api.nomad [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        source      = "${NOMAD_TASK_DIR}/vars-template"
+        destination = "${NOMAD_TASK_DIR}/vars"
+      }
+
+      vault {
+        policies = ["dp-cantabular-dimension-api-publishing"]
+      }
+    }
+  }
+}
```
Comment (redhug1) [inline dp-cantabular-dimension-api.nomad]: Add CR

File: features/helloworld.feature [hunk 8]
```diff
@@ -0,0 +1,7 @@
+Feature: Helloworld
+
+
+
+  Scenario: Posting and checking a response
+    When I GET "/hello"
+    Then I should receive a hello-world response
```
Comment (redhug1) [inline features/helloworld.feature]: Add CR

File: swagger.yaml [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        description: "The last health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_success:
+        type: string
+        description: "The last successful health check date and time of the external service"
+        example: "2020-06-11T11:49:50.330089Z"
+      last_failure:
+        type: string
+        description: "The last failed health check date and time of the external service"
+        example: "2019-09-22T11:48:51.0000001Z"
```
Comment (redhug1) [inline swagger.yaml]: Add CR

File: Dockerfile.local [hunk 10]
```diff
@@ -0,0 +1,9 @@
+FROM golang:1.17-stretch AS base
+
+ENV GOCACHE=/go/.go/cache GOPATH=/go/.go/path TZ=Europe/London
+
+RUN GOBIN=/bin go get github.com/cespare/reflex
+
+# Map between the working directories of dev and live
+RUN ln -s /go /dp-cantabular-dimension-api
+WORKDIR /dp-cantabular-dimension-api
```
Comment (redhug1) [inline Dockerfile.local]: Add CR

**Review by redhug1 — CHANGES_REQUESTED** [review]
Looks fine, apart from the usual github request for additional CR's at end of file ...

File: .gitignore (hunk 1 above)
Comment (ChrysmOre) [reply .gitignore]: All these files come out of code generation without CRs.. amazing

**Review by redhug1 — DISMISSED** [review]

**Review by DavidSubiros — APPROVED** [review]
LGTM and CI stuff has been fixed

**Pull request merged**